### DISCUSSION TITLE
ibm ps/2 model 70 removed (86box build 2983) - why?!?

### DISCUSSION THREAD

**Discussion by honx4amiga** [86Box/86Box#1426] [Q&A] [answered]
why was ibm ps/2 model 70 removed in build 2983? i have one configuration for model 70, because this was my very first pc.
does this mean that i can't update past 2982 anymore if i want to be able to use my ps/2 mod.70 configuration in future?
you already removed a machine in build 2981 and now in 2983 important ps/2 model 70. an "update" feels like downgrade.
so it's not worth downloading "updates" for 86box any longer.

**Comment by ppgrainbow** [comment, marked as answer]
The emulation of the IBM PS/2 Model 70 Type 4 never made it out of dev-branch and was removed, because of the 12903 error at boot, a co-processor error that will eventually never be corrected.

The IBM PS/2 Model 70 Type 4 was replaced with the Olivetti PCS-46C.

The IBM PS/2 Model 70 Type 3 still works. However, you may experience bugs on the emulated machine. If you have found any bugs, please file a bug report. Upon boot for the first time, the Model 70 Type 3 will only show a 162 and 163 error that can be easily corrected.

I hope that this helps. :)

**Reply by dhrdlicka** [reply]
I will just add that 86Box generally prefers diversity and quality over quantity, which is why hardware that doesn't perform satisfactorily under emulation may be removed or replaced. We understand some users might have emotional ties with some of the items that we support, however, our development resources are limited.

Considering that the Type 3 is also emulated and doesn't manifest any of the bugs of the Type 4, we chose to remove support for the Type 4.

**Reply by anikom15** [reply]
Have you considered labeling machines as 'not working' like MAME does? This allows for documenting the behavior and encouraging development while steering casual users away from frustrating bugs.

**Comment by waltje** [comment]
It has long been known that the current x87 emulation is incorrect, because ituses 64bit numbers internally. Since the real chips use 80b internal registers,this creates rounding errors, which is what that POST error is indicating.
So, rather than removing the entire machine, just have it display a message atstartup, telling the user the above "so it can be safely ignored".
Or, rework the x87 to optionally use 80b (16+64, like the x87 does) internally,as an optional ("math_precise") mode, default true on that machine.

**Comment by waltje** [comment]
> I will just add that 86Box generally prefers diversity and quality over quantity.

Yeah, that was made clear over the last year or so, with many, many machines and devicesadded....
--Fred

**Reply by OBattler** [reply]
> Or, rework the x87 to optionally use 80b (16+64, like the x87 does) internally,as an optional ("math_precise") mode, default true on that machine.

Which is in the plans. It will be gotten to eventually. You have no idea how many things I still have to do.

> Yeah, that was made clear over the last year or so, with many, many machines and devicesadded....

You said this, but then below, you say the following:

> So, rather than removing the entire machine, just have it display a message atstartup, telling the user the above "so it can be safely ignored".
> Or, rework the x87 to optionally use 80b (16+64, like the x87 does) internally,as an optional ("math_precise") mode, default true on that machine.

So, should we keep machines regardless of whether or not they even work correctly, with the users being given a message, or should we not have such machines at all?

**Comment by waltje** [comment]
> You have no idea how many things I still have to do.

Oh, I think I have a pretty good idea...

> So, should we keep machines regardless of whether or not they even work correctly, with the users being given a message, or should we not have such machines at all?

The really bad ones (especially those based on more recent guesswork chipsets) should imho beremoved, the others (like the M70.4 should be given a message, similar to the one used with theDEVICE_UNSTABLE bit- just add a MACHINE_ISSUES bit for machines.  That way, you can leavethem in, and users cannot yell at you for an error message (FPU rounding error) or other weirdness.
--Fred.

**Comment by ghost** [comment]
> why was ibm ps/2 model 70 removed in build 2983? i have one configuration for model 70, because this was my very first pc.
> does this mean that i can't update past 2982 anymore if i want to be able to use my ps/2 mod.70 configuration in future?
> you already removed a machine in build 2981 and now in 2983 important ps/2 model 70. an "update" feels like downgrade.
> so it's not worth downloading "updates" for 86box any longer.

That doesn't mean it won't get revisited. The reason why it was removed was because of an unfixable bug on the FPU.

**Comment by TC1995** [comment]
Not to necropost but now softfloat is about to hit the main branch, so we can undev said machine.

**Comment by pankoza2-pl** [comment]
and now it's added back because the new Softfloat FPU that works with this machine has been added to 86Box 😄
